### DISCUSSION TITLE
Allow prefetching SSR Routes

### DISCUSSION THREAD

**Discussion by ttristan** [vercel/next.js#45997] [Ideas]
### Describe the feature you'd like to request

In https://github.com/vercel/next.js/pull/42936 the behaviour of prefetching links was changed in such a way that SSR Routes are still prefetched but return an empty object. This caused some issues in my App when I switched from v12 to 13.

I use the next-pwa package and cache responses. This crashed my App on Navigation and Browser Back when a prefetched response with only an empty object was cached on the client and restored without any pageProps.

I liked prefetching of SSR Routes as it speeds up page transitions and I would like to keep this feature.

### Describe the solution you'd like

Add an additional option alongside `middlewarePrefetch: flexible` to allow prefetching of SSR Routes.

### Describe alternatives you've considered

I tried overriding the `x-middleware-prefetch` request header that was introduced with the change so that SSR pages are not skipped and still prefetched like it was in v12. Apart from the fact that this would probably be a dirty hack I could not make this work.

**Comment by cvbuelow** [comment]
* https://github.com/vercel/next.js/pull/46210
